core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleWriter.java

```diff
@@ -701,4 +735,109 @@ private void wrapLine(boolean space) throws IOException {
 			writer.write(" ");
 		}
 	}
+
+	/*
+	 * not synchronized, assumes calling method has obtained a lock on bufferLock
+	 */
```

Comment:
I wonder if this could be checked in some way. To assert that we're not calling this method from multiple threads.